core/src/main/scala/org/apache/spark/storage/BlockManagerId.scala

```diff
@@ -136,11 +136,14 @@ private[spark] object BlockManagerId {
    * The max cache size is hardcoded to 10000, since the size of a BlockManagerId
    * object is about 48B, the total memory cost should be below 1MB which is feasible.
    */
-  val blockManagerIdCache = CacheBuilder.newBuilder()
-    .maximumSize(10000)
-    .build(new CacheLoader[BlockManagerId, BlockManagerId]() {
-      override def load(id: BlockManagerId) = id
-    })
+  val blockManagerIdCache = {
```

Comment:
nit: maybe change this to:
```scala
  val blockManagerIdCache = Caffeine.newBuilder()
      .maximumSize(10000)
      .build[BlockManagerId, BlockManagerId](identity)
```
?

Reply:
I didn't realize that usage, my Scala level was too poor, thx ~

Reply:
Similar to XXX, it cannot be compiled with `.build[BlockManagerId, BlockManagerId](identity)`

Reply:
7b360a7 revert this change